xsdany.go

```diff
@@ -227,11 +245,23 @@ func (x *XSDAny) MarshalXML(e *xml.Encoder, start xml.StartElement) error {
 
 	// add our prefixes and namespaces to root element
 	for _, pfx := range ns.prefixes {
+	if _, ok := ignorables[pfx]; ok {
+		attrsToIgnore = append(attrsToIgnore, pfx)
+	}
 		ns := ns.prefixToURL[pfx]
 		a.Attrs = append(a.Attrs, xml.Attr{
 			Name:  xml.Name{Local: "xmlns:" + pfx},
 			Value: ns,
 		})
+		if pfx == "mc" {
+			includeIgnorable = true
+		}
+	}
+	if includeIgnorable && len(attrsToIgnore) > 0 {
```

Comment:
Can you add a short comment why we want to ignore those?

Reply:
Word adds mc:Ignorable for all elements which have xmlns:mc attribute when repairing document. I think this is because these attributes relate to different versions of Office and checking of them should be avoided for compatibility reasons.

Reply:
What about in our case?  What are we ignoring?  Are we ignoring them when we read the document or when we write it?  We need to document how this works so our users can understand what to expect

Reply:
Ignorable means that the element can be ignored by versions of Office which is older and can't understand it. It is not mandatory ignorable, this attribute means that it can be ignorable if not understood. This attribute is already contained in source documents in _examples folder, but as it is not  represented in the schema, we have to recreate it from scratch in order to be in the document structure when marshalling it back when saving. So if the element has xmlns:mc attribute which means markup compatibility, we can add elements with prefixes like w14, w15 etc (which means different versions of office) to mc:Ignorable list to tell Word that they can be ignored if it doesn't understand them.

More exactly it is described here https://docs.microsoft.com/en-us/office/open-xml/introduction-to-markup-compatibility#markup-compatibility-in-the-open-xml-file-formats-specification , but I don't know how to comment this. Generally nothing is added to the original document or removed from it, so user will not see changes.

Reply:
Can just say something like the following ?
// Add markup compatibility to instruct older viewers to ignore newer schema.
// See https://docs.microsoft.com/en-us/office/open-xml/introduction-to-markup-compatibility#markup-compatibility-in-the-open-xml-file-formats-specification

> Generally nothing is added to the original document or removed from it, so user will not see changes.
What does that mean?  Seems like this "mc:Ignorable" attribute is added?